lib/ts/recipe/thirdpartypasswordless/recipe.ts

```diff
@@ -0,0 +1,134 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import AuthRecipeWithEmailVerification from "../authRecipeWithEmailVerification";
+import { InputType, NormalisedInputType, PreAndPostAPIHookAction, RecipeInterface } from "./types";
+import EmailVerificationRecipe from "../emailverification/recipe";
+import ThirdPartyRecipe from "../thirdparty/recipe";
+import PasswordlessRecipe from "../passwordless/recipe";
+import { normaliseUserInput } from "./utils";
+import OverrideableBuilder from "supertokens-js-override";
+import RecipeImplementation from "./recipeImplementation";
+import { checkForSSRErrorAndAppendIfNeeded, isTest } from "../../utils";
+import { CreateRecipeFunction, NormalisedAppInfo } from "../../types";
+import DerivedThirdPartyRecipeImplementation from "./recipeImplementation/thirdparty";
+import DerivedPasswordlessRecipeImplementation from "./recipeImplementation/passwordless";
+
+export default class Recipe extends AuthRecipeWithEmailVerification<PreAndPostAPIHookAction, NormalisedInputType> {
+    static instance?: Recipe;
+    static RECIPE_ID = "thirdpartypasswordless";
+
+    recipeImplementation: RecipeInterface;
+    thirdPartyRecipe: ThirdPartyRecipe;
+    passwordlessRecipe: PasswordlessRecipe;
+
+    constructor(
+        config: InputType,
+        recipes: {
+            emailVerification: EmailVerificationRecipe | undefined;
+            thirdParty: ThirdPartyRecipe | undefined;
+            passwordless: PasswordlessRecipe | undefined;
+        }
+    ) {
+        super(normaliseUserInput(config), recipes);
+
+        const builder = new OverrideableBuilder(
+            RecipeImplementation(
+                this.config.recipeId,
+                this.config.appInfo,
+                this.config.preAPIHook,
+                this.config.postAPIHook
+            )
+        );
+
+        const _recipeImplementation = builder.override(this.config.override.functions).build();
+        this.recipeImplementation = _recipeImplementation;
+
+        this.thirdPartyRecipe =
+            recipes.thirdParty === undefined
+                ? new ThirdPartyRecipe(
+                      {
+                          recipeId: this.config.recipeId,
+                          appInfo: this.config.appInfo,
+                          preAPIHook: config.preAPIHook,
+                          postAPIHook: config.postAPIHook,
+                          override: {
+                              emailVerification: config.override?.emailVerification,
+                              functions: function () {
+                                  return DerivedThirdPartyRecipeImplementation(_recipeImplementation);
+                              },
+                          },
+                      },
+                      {
+                          emailVerification: this.emailVerificationRecipe,
```

Comment:
Where is `this.emailVerificationRecipe`? How are the tests passing?

Reply:
The recipe class extends AuthRecipeWithEmailVerification, `this.emailVerificationRecipe` comes from the super class

Reply:
I see. Ok